ocaml/xapi/xapi_vdi.ml

```diff
@@ -362,7 +366,10 @@ let create ~__context ~name_label ~name_description
     | `user -> "user"
     | `rrd -> "rrd"
     | `pvs_cache -> "pvs_cache"
-    | `cbt_metadata -> "cbt_metadata" in
+    | `cbt_metadata -> "cbt_metadata"
+    | `unknown ->
+      error "Unknown type in VDI.create";
+      raise (Api_errors.Server_error(Api_errors.field_type_error,["type"])) in
```

Comment:
I also think that this might be a better choice here than `Api_errors.vdi_incompatible_type`, since `unknown` isn't really a VDI type, this is a slightly different error than using a `cbt_metadata` VDI where it isn't allowed.